command/agent/agent.go

```diff
@@ -429,3 +430,31 @@ func (a *Agent) Stats() map[string]map[string]string {
 	}
 	return output
 }
+
+func (a *Agent) GetConfig() string {
+	a.logger.Printf("[INFO] agent: Dumping config")
+
+	var configCopy *Config
+
+	config_data, _ := json.Marshal(*a.agentConf)
+
+	// Here we unmarshal the data we just dumped
+	// As a hack to copy the agent config
+	// So that we can redact any sensitive data here
+	_ = json.Unmarshal(config_data, &configCopy)
+
+	// Redact sensitive data on our copy of the config
+	configCopy.EncryptKey = Redact(configCopy.EncryptKey)
+	configCopy.KeyringFile = Redact(configCopy.KeyringFile)
+
+	// Now we marshall it again, this time prettier
+	json_config, _ := json.MarshalIndent(configCopy, "", "    ")
+
+	return string(json_config)
+}
+
+func Redact(data string) string {
```

Comment:
Note, by sha256summing empty strings, an attacker would be able to see that there is no encryption key set. Then again, they probably have already figured this out, so I don't think it really matters. Technically though, this is an information leak as the sha256sum of an already empty string is probably well known.